server/memphis_handler_schemas.go

```diff
@@ -0,0 +1,368 @@
+			bson.M{"_id": bson.M{"$in": schema.Versions}},
+		},
+	}
+
+	err = schemasVersionCollection.FindOne(context.TODO(), filter).Decode(&schemaVersion)
+	if err != nil {
+		return models.ExtendedSchemaDetails{}, err
+	}
+
+	extedndedSchemaDetails := models.ExtendedSchemaDetails{
+		ID:            schema.ID,
+		SchemaName:    schema.Name,
+		VersionNumber: schemaVersion.VersionNumber,
+		Active:        schemaVersion.Active,
+		CreatedByUser: schemaVersion.CreatedByUser,
+		Type:          schema.Type,
+		SchemaContent: schemaVersion.SchemaContent,
+	}
+	return extedndedSchemaDetails, nil
+}
+
+func (sh SchemasHandler) GetAllSchemasDetails() ([]models.ExtendedSchema, error) {
+	var schemas []models.Schema
+	cursor, err := schemasCollection.Aggregate(context.TODO(), mongo.Pipeline{
+		bson.D{{"$match", bson.D{{"$or", []interface{}{
+			bson.D{{"is_deleted", false}},
+			bson.D{{"is_deleted", bson.D{{"$exists", false}}}},
+		}}}}},
+	})
+
+	if err != nil {
+		return []models.ExtendedSchema{}, err
+	}
+	if err = cursor.All(context.TODO(), &schemas); err != nil {
+		return []models.ExtendedSchema{}, err
+	}
+
+	var schemaVersion models.SchemaVersion
+	var extednedSchemas []models.ExtendedSchema
+	for _, schema := range schemas {
+		for _, version := range schema.Versions {
+			filter := bson.M{"_id": version}
+			err = schemasVersionCollection.FindOne(context.TODO(), filter).Decode(&schemaVersion)
+			if err != nil {
+				return []models.ExtendedSchema{}, err
+			}
+			if schemaVersion.VersionNumber == 1 {
+				extSchema := models.ExtendedSchema{
+					ID:            schema.ID,
+					Name:          schema.Name,
+					Type:          schema.Type,
+					CreatedByUser: schemaVersion.CreatedByUser,
+					CreationDate:  schemaVersion.CreationDate,
+				}
+				extednedSchemas = append(extednedSchemas, extSchema)
+			}
+		}
+
+	}
+
+	if len(extednedSchemas) == 0 {
+		return []models.ExtendedSchema{}, nil
+	} else {
+		return extednedSchemas, nil
+	}
+}
+
+func (sh SchemasHandler) GetAllSchemaVersionsByVersionNumber(versionNumber int) ([]models.SchemaVersion, error) {
+	var schemasVersion []models.SchemaVersion
+
+	filter := bson.M{"version_number": versionNumber}
+	cursor, err := schemasVersionCollection.Find(context.TODO(), filter)
+	if err != nil {
+		return []models.SchemaVersion{}, err
+	}
+	if err = cursor.All(context.TODO(), &schemasVersion); err != nil {
+		return schemasVersion, err
+	}
+
+	if err != nil {
+		return []models.SchemaVersion{}, err
+	}
+
+	if len(schemasVersion) == 0 {
+		return []models.SchemaVersion{}, nil
+	} else {
+		return schemasVersion, nil
+	}
+}
+
+func (sh SchemasHandler) FindAndDeleteSchema(schemaName string) error {
+	var schema models.Schema
+	filter := bson.M{"name": schemaName}
+	err := schemasCollection.FindOne(context.TODO(), filter).Decode(&schema)
+	if err != nil {
+		return err
+	}
+	schemaVersions := schema.Versions
+	var schemaVersion models.SchemaVersion
+	for _, version := range schemaVersions {
```

Comment:
why using FindOneAndDelete, please use deleteMany it is overkill to access th DB so many times